main.tf

```diff
@@ -105,15 +109,10 @@ resource "aws_iam_role_policy" "autoscaler_cloudwatch" {
   policy = "${data.aws_iam_policy_document.autoscaler_cloudwatch.json}"
 }
 
-data "aws_iam_role" "autoscale_service" {
-  name = "AWSServiceRoleForApplicationAutoScaling_DynamoDBTable"
-}
-
 resource "aws_appautoscaling_target" "read_target" {
   max_capacity       = "${var.autoscale_max_read_capacity}"
   min_capacity       = "${var.autoscale_min_read_capacity}"
   resource_id        = "table/${module.default.id}"
-  role_arn           = "${data.aws_iam_role.autoscale_service.arn}"
```

Comment:
@osterman 
it was like this before.
you changed it to use `AWSServiceRoleForApplicationAutoScaling_DynamoDBTable ` because of
> I had to use that to get idempotent apply

Reply:
Ya, exactly. Everytime I ran `terraform apply` there was a change set. 

@darend what happens if you run it twice?

Reply:
This was fixed in a newer AWS provider: https://github.com/terraform-providers/terraform-provider-aws/pull/3303

Reply:
![image](https://user-images.githubusercontent.com/1302480/37536708-a4065242-2908-11e8-8bea-0c96ccc60490.png)